deploy/util/configvalidation.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+import yamale # type: ignore
+
+from typing import Optional, List
+
+def validate(src_yaml: str, schema_yaml: str, extra_schema_yaml: Optional[str] = None, ignore_statements: List[str] = [], src_as_str: bool = False) -> bool:
+    """Validate a src yaml file with a schema file (or extra one)
+
+    Args:
+        src_yaml: path to yaml file (default) or yaml string to check
+        schema_yaml: path to schema yaml file to use for check
+        extra_schema_yaml: path to another schema yaml file to use for check
+        ignore_statements: if an error message matches any string in the list of strings then ignore that error
+        src_as_str: treat src_yaml input as a yaml string instead of file
```

Comment:
nit: i'd personally prefer to make the file-based function call the string based one after parsing, but YMMV.

Reply:
Let me see if I can do that... However, I'm not sure Yamale nicely provides a function to parse the YAML string given

Reply:
Solving this will require a bit more duplication than I personally think would be good to do. I would have to duplicate this code and also roll my own `make_data` function:

https://github.com/23andMe/Yamale/blob/d7f96574b779a66dc25085cc172182f8b7f351db/yamale/readers/yaml_reader.py#L26-L37

https://github.com/23andMe/Yamale/blob/d7f96574b779a66dc25085cc172182f8b7f351db/yamale/yamale.py#L27-L32

For now, I can clean up the API so that there is a `src_path` and `src_str` param that are separated.